index.js

```diff
@@ -59,14 +60,20 @@ function getCallerFilename () {
 
 function installGlobally (toLoad, mocks) {
   var callerFilename = getCallerFilename()
+  var parent = require.cache[toLoadPath] || null
 
   // Inject all of our mocks
   Object.keys(mocks).forEach(function (name) {
     var namePath = resolve(callerFilename, name)
     if (mocks[name] == null) {
       delete require.cache[namePath]
     } else {
-      require.cache[namePath] = {exports: mocks[name]}
+      var old = require.cache[namePath]
+      var mod = new Module(namePath, null)
+      mod.filename = namePath
```

Comment:
The reason `new Module(namePath, null)` is called with `null` for the `parent` param is because if it's provided there then the `parent.children` collection would be modified _(with the mock module added to it)_. Adding it after the `new Module` call via `mod.parent =` avoids tracking the mock in the parent.